### ISSUE TITLE
Components from templates are not themed

### ISSUE THREAD

**Issue by anssit** [vaadin/flow#3907]
Add e.g. a `vaadin-combo-box` into `example-template.html` in https://vaadin.com/start/v10-project-base with Vaadin Designer so that in Designer the template looks like this:

![screen shot 2018-04-17 at 8 47 09](https://user-images.githubusercontent.com/16914037/38851011-8ffedce2-421c-11e8-955e-7f64c6902973.png)

Corresponding source:

```html
<!-- Dependency resources -->
<link rel="import" href="../bower_components/polymer/polymer-element.html">
<link rel="import" href="../bower_components/vaadin-combo-box/src/vaadin-combo-box.html">

<!-- Defines the example-template element -->
<dom-module id="example-template">
    <template>
<span>[[value]]</span>
 <vaadin-combo-box items="[[items]]"></vaadin-combo-box>
</template>

    <!-- Polymer boilerplate to register the example-template element -->
    <script>
        class ExampleTemplate extends Polymer.Element {
            static get is() {
                return 'example-template'
            }
        }
        customElements.define(ExampleTemplate.is, ExampleTemplate);
    </script>
</dom-module>
```

Result when running the application (`mvn jetty:run`):

<img width="353" alt="screen shot 2018-04-17 at 8 47 13" src="https://user-images.githubusercontent.com/16914037/38851030-a8db2f72-421c-11e8-9df5-9e7c65338d03.png">

Vaadin Platform 10.0.0.beta8.

**Comment by caalador** [comment]
A quick check on the issue is that I see
"/frontend/theme/lumo/../bower_components/polymer/polymer-element.html"

That means we have at least 2 problems.
1. all instances of `/src/` are changed when only the last one should be.
2. the relative path makes it so that we won't find a file in webjars as now we start with `/frontend/src/.../bower_components/...` when it should be `/frontend/bower_components/...`

I'll look into this.

**Comment by Artur-** [comment]
Have you checked it is not fixed as a side effect of introducing a new resolving API in VaadinService? (done today and still needs some more PRs)

**Comment by caalador** [comment]
If you mean your `normalize-uris` branch then that should basically fix the issue 2, but I already had that part in my branch.
Also do you dig deep enough with the normalize e.g. if the relative is `../../../my/component` as this is the case in bakery.

**Comment by caalador** [comment]
I made the PR with only the theme fix for part 1.
This ticket should be fixed after #3916 is also merged, but needs to be tested and needs added validation tests.

**Comment by caalador** [comment]
Tested with bakery and skeleton, this now works as it should.
